pytest_rerunfailures.py

```diff
@@ -160,27 +185,105 @@ def get_reruns_count(item):
+
+
+# Constant delay among reruns.
+class RerunDelayConst(RerunDelay):
+    def __init__(self, delay):
+        self._delay = delay
+
+    def next_rerun_delay(self, _):
+        return self._delay
+
+
+# Random exponential backoff delay.
+class RerunDelayExp(RerunDelay):
+    def __init__(self, factor, base, max):
+        self._factor = factor
+        self._base = base
+        self._max = max
+
+    def next_rerun_delay(self, iteration_nr):
+        upper = self._factor * (self._base**iteration_nr)
+        if self._max:
+            upper = min(self._max, upper)
+
+        return random.randint(self._factor, upper)
+
+
 def get_reruns_delay(item):
     rerun_marker = _get_marker(item)
 
     if rerun_marker is not None:
         if "reruns_delay" in rerun_marker.kwargs:
             delay = rerun_marker.kwargs["reruns_delay"]
         elif len(rerun_marker.args) > 1:
-            # check for arguments
+            # check for argument
             delay = rerun_marker.args[1]
         else:
             delay = 0
+
+        if "reruns_delay_backoff" in rerun_marker.kwargs:
+            delay_backoff = rerun_marker.kwargs["reruns_delay_backoff"]
+        elif len(rerun_marker.args) > 2:
+            # check for argument
+            delay_backoff = rerun_marker.args[2]
+        else:
+            delay_backoff = False
+
+        if "reruns_delay_backoff_base" in rerun_marker.kwargs:
+            delay_backoff_base = rerun_marker.kwargs["reruns_delay_backoff_base"]
+        elif len(rerun_marker.args) > 3:
+            # check for argument
+            delay_backoff_base = rerun_marker.args[3]
+        else:
+            delay_backoff_base = DELAY_BACKOFF_BASE_DEFAULT
+
+        if "reruns_delay_backoff_max" in rerun_marker.kwargs:
+            delay_backoff_max = rerun_marker.kwargs["reruns_delay_backoff_max"]
+        elif len(rerun_marker.args) > 4:
+            # check for argument
+            delay_backoff_max = rerun_marker.args[4]
+        else:
+            delay_backoff_max = None
+
     else:
         delay = item.session.config.option.reruns_delay
 
+        delay_backoff = item.session.config.option.reruns_delay_backoff
+
+        delay_backoff_base = \
+            item.session.config.option.reruns_delay_backoff_base
+
+        delay_backoff_max = item.session.config.option.reruns_delay_backoff_max
+
     if delay < 0:
         delay = 0
         warnings.warn(
             "Delay time between re-runs cannot be < 0. Using default value: 0"
         )
 
-    return delay
+    if delay_backoff_base < 0:
+        delay_backoff_base = DELAY_BACKOFF_BASE_DEFAULT
+        warnings.warn(
+            "Exponential backoff base cannot be < 0. Using default value "
+            f"{DELAY_BACKOFF_BASE_DEFAULT}"
+        )
+
+    if delay_backoff_max and delay_backoff_max < 0:
+        delay_backoff_max = None
+        warnings.warn(
+            "Max value for exponential backoff cannot be < 0. "
+            "Do not enforce it"
```

Comment:
I suggest `"Ignoring."` or `"Not enforcing it."`

Reply:
Done